File: codes/cpp/Code_5_10_3.cpp
#include <iostream>
using namespace std;

long long N, X, Y;

int main() {
	// 入力
	cin >> N >> X >> Y;

	// 4 つの整数 (a, b, c, d) の全探索 → 答えの出力
	for (int a = 1; a <= N; a++) {
		for (int b = a; b <= N; b++) {
			for (int c = b; c <= N; c++) {
				for (int d = c; d <= N; d++) {
					if (a + b + c + d == X && 1LL * a * b * c * d == Y) {
						cout << "Yes" << endl;
						return 0; // プログラムの実行を終了させる
					}
				}
			}
		}
	}
	cout << "No" << endl; // 1 つも見つからなかったら No
	return 0;
}

File: codes/python/Code_5_10_3.py
# 入力
N, X, Y = map(int, input().split())

# 4 つの整数 (a, b, c, d) の全探索
flag = False
for a in range(1, N + 1):
	for b in range(a, N + 1):
		for c in range(b, N + 1):
			for d in range(c, N + 1):
				if a + b + c + d == X and a * b * c * d == Y:
					flag = True  # 答えが見つかったら flag を true にする

# 答えの出力
if flag == True:
	print("Yes")
else:
	print("No")

# p.256 注 5.10.2 に書かれてあるように、Python は実行速度が遅いので、最大ケースでは実行に数十秒かかります。
# しかし、同じプログラムを PyPy3 で実行すると、最大ケースでも 1 秒ほどで答えを求めることができます。
# より良い計算量のプログラムについては、Code_5_10_3_extra.py を見てください。
